utils/firestore.js

```diff
@@ -2152,16 +2152,29 @@ const updateKitAssemblyData = async (data) => {
     }
 }
 
-const checkCollectionUniqueness = async (supplyId, collectionId) => {
+const checkCollectionUniqueness = async (supplyId, collectionId, returnKitTrackingNumber) => {
     try {
         const supplySnapShot = await db.collection('kitAssembly').where('690210658', '==', supplyId).get();
         const collectionSnapShot = await db.collection('kitAssembly').where('259846815', '==', collectionId).get();
-        if (supplySnapShot.docs.length === 0 && collectionSnapShot.docs.length === 0) {
+        let returnKitTrackingNumberSnapshot = {docs: []};
+        let supplyKitTrackingNumberSnapshot = {docs: []};
+        if(returnKitTrackingNumber) {
```

Comment:
The `returnKitTrackingNumber` can be checked earlier.

Reply:
Assuming bad value or missing values of `returnKitTrackingNumber` are passed from API calls, it can be a little more performant to do checks at API entry point. But current way works.